packages/tooling/fast-tooling-react/src/form/custom-controls/css/color-picker-component/color-picker.ts

```diff
@@ -0,0 +1,400 @@
+        this.currentMouseParam = null;
+        this.isMouseActive = false;
+    }
+
+    /**
+     * Handles changes to any of the color property text inputs typed by the user.
+     * @param param ['r','g','b','a','h','s','v'] - String specifying which color value is being modified.
+     * @param e - Reference to the event.
+     */
+    public handleTextValueInput(param:string,e:Event)
+    {
+        const inputVal = (<HTMLInputElement>e.composedPath()[0]).value;
+        if(isNullOrWhiteSpace(inputVal) || Number.isNaN(inputVal))
+        {
+            return;
+        }
+        let newVal: number = parseInt(inputVal);
+
+        if(['r','g','b','a'].includes(param))
+        {
+            if( (param != 'a' && this.isValideRGB(newVal)) || ( param == 'a' && this.isValideAlpha(newVal) ))
+            {
+                this.currentRGBColor = new ColorRGBA64(
+                    param == 'r' ? newVal / 255 : this.currentRGBColor.r,
+                    param == 'g' ? newVal / 255 : this.currentRGBColor.g,
+                    param == 'b' ? newVal / 255 : this.currentRGBColor.b,
+                    param == 'a' ? newVal / 100 : this.currentRGBColor.a
+                );
+                this.currentHSVColor = rgbToHSV(this.currentRGBColor);
+                this.updateUIValues(true);
+            }
+        }
+        else if(['h','s','v'].includes(param))
+        {
+            if( (param != 'h' && this.isValideSatVal(newVal)) || ( param == 'h' && this.isValideHue(newVal) ))
+            {
+                this.updateHSV(
+                    param == 'h' ? newVal : this.currentHSVColor.h,
+                    param == 's' ? newVal / 100 : this.currentHSVColor.s,
+                    param == 'v' ? newVal / 100 : this.currentHSVColor.v
+                );
+            }
+        }
+    }
+
+    /**
+     * Change event handler for inner control.
+     * @remarks
+     * "Change" events are not `composable` so they will not
+     * permeate the shadow DOM boundary. This fn effectively proxies
+     * the change event, emitting a `change` event whenever the internal
+     * control emits a `change` event
+     * @internal
+     */
+    public handleChange(): void 
+    {
+        this.$emit("change");
+    }
+
+    /**
+     * Determines if a number value is within the valid range for an R, G, or B color channel.
+     * @param val - Number to be evaluated.
+     */
+    private isValideRGB(val: number): boolean
+    {
+        return val >= 0 && val <= 255;
+    }
+
+    /**
+     * Determines if a number value is within the valid range for the alpha channel.
+     * @param val - Number to be evaluated.
+     */
+    private isValideAlpha(val: number): boolean
+    {
+        return val >= 0 && val <= 100;
+    }
+
+    /**
+     * Determines if a number value is within the valid range for the saturation or value color channels.
+     * @param val - Number to be evaluated.
+     */
+    private isValideSatVal(val: number): boolean
```

Comment:
Spelling on "valid", also we prefer to use the full text wherever possible since this will get minified later anyway.

```suggestion
    private isValidSaturationValue(value: number): boolean
```

Reply:
Done